pyscf/mcpdft/otfnal.py

```diff
@@ -32,6 +32,57 @@
 FT_C = getattr(__config__, 'mcpdft_otfnal_ftransfnal_C', -85.38149682)
 
 OT_HYB_ALIAS = {'PBE0' : '0.25*HF + 0.75*PBE, 0.25*HF + 0.75*PBE'}
+REG_FUNCTIONALS={}
+
+def register_ot_libxc(xc_base):
+    '''
+    This is a wrapper function to registers the new ot-functionals if they haven't been
+    registered previously.
+    Args:
+        xc_base: str
+            The name of the ot-functional to be registered.
+    '''
+    xc_base = xc_base.replace("-", "") # In case, the functional name has a hyphen
+
+    def _register_repM06L():
+        '''
+        This function register the reparametrized
+        M06-L functional (used for MC23 Functional) in the libxc library.
+        '''
+        from pyscf import dft2
+        dft.libxc = dft2.libxc
+        dft.numint.libxc = dft2.libxc
+        dft.numint.LibXCMixin.libxc = dft2.libxc
```

Comment:
Could we perhaps do this a little more cleanly? Can we make it so that the entire MC-PDFT module just uses `dft2.libxc` at all times, even when we aren't building these new functionals? Perhaps by intervening in `get_transfnal` and trying to copy some of the logic in `mcdcft.dcfnal.dcfnal.__init__`? This is a highly, highly impure function and it makes me very nervous.

Reply:
I am also in favor of this suggestion. As a first step, I think the easiest way to avoid impure function is probably override the `libxc` attribute of the `_numint` instance after it is first initialized in the constructors of the on-top functional classes. In the long run, perhaps it would be a good idea to completely remove the `ks` objects from the MC-PDFT module, since all we need is `dft.numint.NumInt` and `dft.grid.Grids`.